lib/identity/client.js

```diff
@@ -602,11 +561,69 @@ class Client extends PartialClient {
       currentGroup.groupID,
       recordType
     )
-    if (secretType === 'File') {
-      await this.storage.share(recordType, clientID)
-    }
     return sharedWithGroup.record_type
   }
+  /**
+   * addSecretToNamespace takes a user defined namespace name and a secret to add to the namespace
+   *
+   * @param {String} namespace The user given name for the secret stored in the meta data
+   * @param {String} secretType The secret type, example 'Credential'
+   * @param {String} secretName The user given name for the secret stored in the meta data
+   *
+   * @return {Boolean}
+   */
+  async addSecretToNamespace(secretName, secretType, namespace) {
+    // Check if the namespace exists
+    let currentGroup = await this.getOrCreateNamespace(namespace)
+    // Generate the Record Type to share
+    const recordType = `tozny.secret.${SECRET_UUID}.${secretType}.${secretName}`
+    let sharedWithGroup = await this.storage.shareRecordWithGroup(
+      currentGroup.groupID,
+      recordType
+    )
+    return sharedWithGroup.record_type
+  }
+  /**
+   * addIdentityToNamespace takes a user defined namespace name and a username for an identity and gives access to the given namespace for the identity
+   *
+   * @param {String} namespace The user given name for the secret stored in the meta data
+   * @param {String} usernameToAdd The username of the user that will have permissions added
+   *
+   * @return {Boolean}
+   */
+  async addIdentityToNamespace(usernameToAdd, namespace) {
+    // Validation
+    if (usernameToAdd == '') {
+      throw new Error('Username Required')
+    }
+    // Look Up the ClientID for Username
+    let clientID = await this.clientIDLookUpByUsername(usernameToAdd)
+    if (clientID === null) {
+      return null
+    }
+    // Check if namespace exists
+    const groupName = `tozny.secret.${this.config.realmName}.${namespace}`
+    const groupList = await this.storage.listGroups(
+      this.storage.config.clientId,
+      [groupName]
+    )
+    if (groupList.groups.length < 1) {
```

Comment:
DId the `getNamespace` method get refactored out?

This method should do the same `getOrCreateNamespace` but I'd like to adjust all of it to a single helper method which can perform the `getNamespace` or `getOrCreateNamescpace` based on call params:

```ts
function getNamespace(namespace: string, createIfMIssing: boolean){
...
}
```

This kind of signature.